### PR TITLE
Add `istioctl collateral --zsh` to generate zsh completion files

### PR DESCRIPTION
**Pull request by ericvn**
The current zsh completion file genereated via cobra doesn't work.
  Waiting for them to merge their PR https://github.com/spf13/cobra/pull/828 to get actual working file.

### CODE REVIEW

**Comment by ericvn** [comment]
@esnible This is the PR to add the `collateral --zsh` parameter and have it call to cobra to generate the file. Once they update their code to generate an acceptably working _zsh_ completion file we can move beyond this with additional PRs for the other bullets in the issue: https://github.com/istio/istio/issues/13139

File: pkg/collateral/control.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ type Control struct {
 	// EmitBashCompletion controls whether to produce bash completion files.
 	EmitBashCompletion bool
 
+	// EmitZshCompletion controls whether to produce bash completion files.
```
Comment (esnible) [inline pkg/collateral/control.go]: Comment should be "zsh completion files"?

Comment (ericvn) [reply pkg/collateral/control.go]: Done.

**Comment by esnible** [comment]
Does this require the unmerged Cobra patch?

**Comment by ericvn** [comment]
@esnible It only requires the Cobra patch if you want to be able to actually use the file generated by the `istioctl` command. The current cobra code supports the call to generate the file, the file just doesn't work correctly. Once the cobra PR is merged, and pull that new version in and rebuild, we can use the file without any further code changes on our part. Hopefully, that makes sense.

**Comment by esnible** [comment]
Marking as `do-not-merge` because it is blocked by needing the Cobra patch to merge and a Gopkg referring to a Cobra version with the patch.

**Comment by howardjohn** [comment]
Please make the same change we made for bash too:

https://github.com/istio/istio/issues/13564#issuecomment-486276567

otherwise it will break osx

**Comment by ericvn** [comment]
Based on feedback in https://github.com/istio/istio/issues/13139, I will create a new PR based on the work in Kubernetes, helm, cobra with wrapping the `bash` completion output. That will be in a different PR.

**Pull request closed without merging**